### ISSUE TITLE
IPNS seems down

### ISSUE THREAD

**Issue by wmitsuda** [ethereum/sourcify#660]
it seems something is down again, I can't resolve the ipns name:

```
time ipfs resolve /ipns/k51qzi5uqu5dll0ocge71eudqnrgnogmbr37gsgl12uubsinphjoknl6bbi41p
Error: could not resolve name
ipfs resolve   0.23s user 0.04s system 0% cpu 1:00.05 total
```

The public gateway at [ipfs.io](https://gateway.ipfs.io/ipns/k51qzi5uqu5dll0ocge71eudqnrgnogmbr37gsgl12uubsinphjoknl6bbi41p) also can't resolve it.

Not sure if it is just the name or the entire service is down.

_Originally posted by @wmitsuda in https://github.com/ethereum/sourcify/issues/495#issuecomment-1040981026_

**Comment by kuzdogan** [comment]
Yes unfortunately we had to stop the ipfs service on the server as it is causing an out-of-memory and crashes the servers.

We're regularly restarting the ipfs container as a workaround and it seems to work for now on the staging server. I will push it to production today.

**Comment by kuzdogan** [comment]
As a sum up and update, the past issues have been the following:
- We moved the Sourcify directory on our production server into the larger HDD as we completely ran out of storage and it was even stopping system processes. As a result read operations in `ipfs add` started to take ages and we were never able to pin
- As we were doing an `ipfs add` every time, it was duplicating the whole repo under `.ipfs` and this was causing our SSD to wear out. To avoid duplication I was trying out the ipfs Filestore (`--nocopy` flag on `add`). However it requires file contents not to change at all. Our `manifest.json` file's "timestamp" field is updated everytime a contract is added and this causes the `ipfs add --nocopy /repository` to fail. 
- IPFS was using sometimes too much memory and CPU which I could not figure out why.

Here's what's planned:
- As a workaround to Filestore issue, I am trying to pin only `/repository/contracts` and manually link the `/contracts` under the root CID. 
- We are moving all of Sourcify to a new server. This might take a while to build and migrate. With being back to an SSD, it should work on the production server.
- The new server should be able to provide enough resources to IPFS nodes. 

The staging publishing works and available at https://ipfs.ethdevops.io/ipns/k51qzi5uqu5dkuzo866rys9qexfvbfdwxjc20njcln808mzjrhnorgu5rh30lb

It's better to use now the Filestore --nocopy but still there should be a better way to update the root hash with newly added contracts. I'm leaning towards calling `ipfs add` and MFS `ipfs files cp` over the IPFS API each time a contract is added.

**Comment by kuzdogan** [comment]
Resolved with the migration to the new server and https://github.com/ethereum/sourcify/pull/702. 

Feel free to open this or a new issue if any new problems come up
